spec/migrations/20211119152213_remove_hawkular_authentications_spec.rb

```diff
@@ -0,0 +1,47 @@
+require_migration
+
+class RemoveHawkularAuthentications < ActiveRecord::Migration[6.0]
+  class ExtManagementSystem < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    has_many :authentications, :as => :resource
+    has_many :endpoints, :as => :resource
+  end
+end
+
+describe RemoveHawkularAuthentications do
+  let(:auth_stub)     { migration_stub(:Authentication) }
+  let(:endpoint_stub) { migration_stub(:Endpoint) }
+  let(:ems_stub)      { migration_stub(:ExtManagementSystem) }
+
+  migration_context :up do
+    let(:kube) { ems_stub.create!(:type => "ManageIQ::Providers::Kubernetes::ContainerManager", :name => "kube") }
+
+    before do
+      auth_stub.create!(:type => "AuthToken", :authtype => "bearer", :userid => "_", :resource_id => kube.id, :resource_type => ems_stub)
```

Comment:
Might be better to use kube.class here, to keep it consistent.

```suggestion
      auth_stub.create!(:type => "AuthToken", :authtype => "bearer", :userid => "_", :resource_id => kube.id, :resource_type => kube.class)
```

Reply:
Just habit from `kube.class` usually being the namespaced subclass where resource_type is the base class.  It does work in this case though so updated